File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/StatisticsSample/src/com/hp/workpath/sample/statisticsample/view/common/DigitalSendInfoView.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.statisticsample.view.common;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.hp.workpath.api.statistics.jobinfo.StatisticsAttributes;
import com.hp.workpath.sample.statisticsample.R;
import com.hp.workpath.sample.statisticsample.view.Utils;

public class DigitalSendInfoView {

    LinearLayout rootView;
    View view;
    FileInfoView fileInfoView;
    FileInfoView fileInfoMetadataView;

    ViewGroup layoutResult;
    ViewGroup layoutTotalDataSize;

    LinearLayout layoutFiles;
    LinearLayout layoutMetadataFile;

    public DigitalSendInfoView(LayoutInflater inflater, LinearLayout rootView) {
        this.rootView = rootView;
        this.view = inflater.inflate(R.layout.layout_digital_send_info, rootView, false);
        initViewDigitalSendInfo();
        initViewClass(inflater);
    }

    public void setDigitalSendInfo(StatisticsAttributes.DigitalSendInfo digitalSendInfo) {
        rootView.removeAllViews();
        if (digitalSendInfo != null) {
            setDeliveredFile(digitalSendInfo.getDeliveredFiles());
            Utils.setSummary(layoutResult, digitalSendInfo.getResult());
            Utils.setSummary(layoutTotalDataSize, digitalSendInfo.getTotalDataSize());
            rootView.addView(view);
        } else {
            LinearLayout parent = (LinearLayout) rootView.getParent();
            parent.setVisibility(View.GONE);
        }
    }

    public void setDeliveredFile(StatisticsAttributes.DigitalSendInfo.DeliveredFile deliveredFile) {
        if (deliveredFile != null) {
            fileInfoView.setFileInfo(deliveredFile.getFiles());
            fileInfoMetadataView.setFileInfo(deliveredFile.getMetdataFile());
        } else {
            LinearLayout parent = (LinearLayout) view.findViewById(R.id.titleDeliveredFilesTextView).getParent();
            parent.setVisibility(View.GONE);
        }
    }

    private void initViewDigitalSendInfo() {
        ((TextView) view.findViewById(R.id.titleDeliveredFilesTextView)).setText(R.string.deliveredFiles);
        layoutResult = Utils.setTitle(view.findViewById(R.id.layoutResult), R.string.result);
        layoutTotalDataSize = Utils.setTitle(view.findViewById(R.id.layoutTotalDataSize), R.string.totalDataSize);
        layoutFiles = Utils.getLayout(view.findViewById(R.id.layoutFiles), R.string.files);
        layoutMetadataFile = Utils.getLayout(view.findViewById(R.id.layoutMetadataFile), R.string.metadataFile);
    }

    private void initViewClass(LayoutInflater inflater) {
        fileInfoView = new FileInfoView(inflater, layoutFiles);
        fileInfoMetadataView = new FileInfoView(inflater, layoutMetadataFile);
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/StatisticsSample/src/com/hp/workpath/sample/statisticsample/view/common/DigitalSendInfoView.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.statisticsample.view.common

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.hp.workpath.api.statistics.jobinfo.StatisticsAttributes.DigitalSendInfo
import com.hp.workpath.api.statistics.jobinfo.StatisticsAttributes.DigitalSendInfo.DeliveredFile
import com.hp.workpath.sample.statisticsample.R
import com.hp.workpath.sample.statisticsample.view.Utils
import com.hp.workpath.sample.statisticsample.view.Utils.getLayout
import com.hp.workpath.sample.statisticsample.view.Utils.setTitle

class DigitalSendInfoView(inflater: LayoutInflater, var rootView: LinearLayout) {
    var view: View = inflater.inflate(R.layout.layout_digital_send_info, rootView, false)
    private lateinit var fileInfoView: FileInfoView
    private lateinit var fileInfoMetadataView: FileInfoView

    private lateinit var layoutResult: ViewGroup
    private lateinit var layoutTotalDataSize: ViewGroup
    private lateinit var layoutFiles: LinearLayout
    private lateinit var layoutMetadataFile: LinearLayout

    fun setDigitalSendInfo(digitalSendInfo: DigitalSendInfo?) {
        rootView.removeAllViews()
        if (digitalSendInfo != null) {
            setDeliveredFile(digitalSendInfo.deliveredFiles)
            Utils.setSummary(layoutResult, digitalSendInfo.result)
            Utils.setSummary(layoutTotalDataSize, digitalSendInfo.totalDataSize)
            rootView.addView(view)
        } else {
            val parent = (rootView as ViewGroup).parent as LinearLayout
            parent.visibility = View.GONE
        }
    }

    private fun setDeliveredFile(deliveredFile: DeliveredFile?) {
        if (deliveredFile != null) {
            fileInfoView.setFileInfo(deliveredFile.files)
            fileInfoMetadataView.setFileInfo(deliveredFile.metdataFile)
        } else {
            val parent =
                view.findViewById<View>(R.id.titleDeliveredFilesTextView).parent as LinearLayout
            parent.visibility = View.GONE
        }
    }

    private fun initViewDigitalSendInfo() {
        (view.findViewById<View>(R.id.titleDeliveredFilesTextView) as TextView).setText(R.string.deliveredFiles)
        layoutResult = setTitle(view.findViewById(R.id.layoutResult), R.string.result)
        layoutTotalDataSize =
            setTitle(view.findViewById(R.id.layoutTotalDataSize), R.string.totalDataSize)
        layoutFiles = getLayout(view.findViewById(R.id.layoutFiles), R.string.files)
        layoutMetadataFile =
            getLayout(view.findViewById(R.id.layoutMetadataFile), R.string.metadataFile)
    }

    private fun initViewClass(inflater: LayoutInflater) {
        fileInfoView = FileInfoView(inflater, layoutFiles)
        fileInfoMetadataView = FileInfoView(inflater, layoutMetadataFile)
    }

    init {
        initViewDigitalSendInfo()
        initViewClass(inflater)
    }
}